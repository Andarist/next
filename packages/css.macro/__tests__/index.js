// @flow
import tester from 'macro-tester'

tester('@emotion/css.macro', __dirname)
