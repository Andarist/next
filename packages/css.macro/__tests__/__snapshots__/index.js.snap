// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@emotion/css.macro actual-expected-usage 1`] = `
"/** @jsx jsx */
import jsx from '@emotion/jsx'
import css from '@emotion/css.macro'

const SomeComponent = () => (
  <div
    css={css\`
      color: hotpink;
    \`}
  />
)


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

/** @jsx jsx */
import jsx from '@emotion/jsx';

var _ref = _css\`
      color: hotpink;
    \`;

const SomeComponent = () => <div css={_ref} />;"
`;

exports[`@emotion/css.macro basic 1`] = `
"import css from '@emotion/css.macro'

function doThing() {
  return css\`
    display: flex;
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

var _ref = _css\`
    display: flex;
  \`;

function doThing() {
  return _ref;
}"
`;

exports[`@emotion/css.macro call-expression 1`] = `
"import css from '@emotion/css.macro'

function doThing() {
  return css({ color: 'hotpink' })
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

var _ref = _css({
  color: 'hotpink'
});

function doThing() {
  return _ref;
}"
`;

exports[`@emotion/css.macro impure 1`] = `
"import css from '@emotion/css.macro'

function doThing() {
  return css\`
    display: \${thing()};
  \`
}


      ↓ ↓ ↓ ↓ ↓ ↓

import _css from \\"@emotion/css\\";

function doThing() {
  return _css\`
    display: \${thing()};
  \`;
}"
`;
